api/src/database/games/Games.ts

```diff
@@ -26,22 +36,32 @@ export const createGame = async (
     owners?: string[],
     moderators?: string[],
 ) => {
-    if (await gameForSlug(slug)) {
-        return null;
-    }
-    return prisma.game.create({
-        data: {
-            name,
-            slug,
-            coverImage,
-            owners: {
-                connect: owners?.map((o) => ({ id: o })),
+    let game : DbGame | null = null;
+    try {
+        game = await prisma.game.create({
+            data: {
+                name,
+                slug,
+                coverImage,
+                owners: {
+                    connect: owners?.map((o) => ({ id: o })),
+                },
+                moderators: {
+                    connect: moderators?.map((m) => ({ id: m })),
+                },
             },
-            moderators: {
-                connect: moderators?.map((m) => ({ id: m })),
-            },
-        },
-    });
+        });
+    } catch (error: unknown) {
+        if (error instanceof Prisma.PrismaClientKnownRequestError) {
+            if (error.code === 'P2002') {
+                return {statusCode: 400, message: "Game with this slug already exists"}
+            }
+            logError(`Database Known Client error - ${error.message}`);
+            return {statusCode: 500, message: "Database error"}
+        }
+        logError(`Database Unknown error - ${error}`);       
+    }
+    return game;
```

Comment:
You should be able to just return the game from the try block which also avoids the need for the interface definition